storage loader

BloamBla opened the pull request:
- when you start your app, goals from local storage added automaticaly.

BloamBla committed:
storage laoder

- when you start your app, goals from local storage added automaticaly.

BloamBla committed:
some changes

- now your completed goals save in local storage to, and when you load the app all completed goals from local storage add to your list.

BloamBla committed:
done goals

- done goals can be saved in local storage and load from.

BloamBla committed:
local store

- was changed type of using local store(now you can choose what you want to change, and after that you need click button to save or load).

BloamBla committed:
bug fix

- on laod app all goals from local storage add in list.

burymm commented on src/index.js:
```diff
@@ -32,13 +32,10 @@ angular.module('ToDo').controller('todoController',
 
             $scope.onLoad = function () {
-                let arr = JSON.parse(localStorage.getItem('todos'));
+                const arr = JSON.parse(localStorage.getItem('allTodos'));
```
You can use destruction like
```
  const [todos, compleateTodos] = JSON.parse(localStorage.getItem('allTodos'));
 
ctrlConnect.setTodos(todos);
 ctrlConnect.setComplTodos(compleateTodos);
```

burymm commented on src/controllers/local-storage-page-controller.js:
```diff
@@ -1,39 +1,75 @@
-                        todosMemoryValue() {
-                            return ctrlConnect.getTodos();
+                        memoryValue() {
+                            const arr = [];
```
You use this code in several places. May be good idea make some function to simplify code?
```
const arr = [];
arr[0] = ctrlConnect.getTodos();
arr[1] = ctrlConnect.getComplTodos();
return arr;
```

burymm commented on src/controllers/local-storage-page-controller.js:
```diff
@@ -46,34 +82,80 @@ angular.module('ToDo').controller('localStoragePageController',
-                            res = ctrlConnect.getTodos().concat(JSON.parse(localStorage.getItem('todos')));
-                            $scope.setInService(res);
+                        if (localStorage.getItem('allTodos') !== null) {
+                            res[0] = JSON.parse(localStorage.getItem('allTodos'))[0];
```
load and parse json file twice not optimize decision. You can use destructuring for that.

burymm commented on src/controllers/main-page-controller.js:
```diff
@@ -51,7 +51,9 @@ angular.module('ToDo').controller('mainPageController', ['$scope', '$uibModal',
             arr = $scope.completetodos;
+            if (!arr) {arr = [];}
             ctrlConnect.setComplTodos(arr);
+            /*localStorage.setItem('completetodos', JSON.stringify(arr));*/
```
looks like legacy code

burymm commented on src/controllers/save-to-loc-contr.js:
```diff
@@ -1,11 +1,24 @@
-        $scope.pieces = JSON.parse(localStorage.getItem('todos'));
+        switch (obj) {
+            case 'allTodos':
+                $scope.pieces[0] = JSON.parse(localStorage.getItem(obj))[0];
```
the same issue with destructuring

burymm commented on styles/main.scss:
```diff
@@ -277,18 +341,13 @@ div.to-do-wrap > div.btn-group > .sort-button-active:active {
     margin: 0 auto;
-    display: flex;
-}
+    padding: 10px}
```
semicolon  missing

BloamBla committed:
bug fix

- if was used 'todos' or 'completetodos' names in local store all goals from this names will be saved;
- some bugs fixed.

BloamBla committed:
Destructuring some arrays

burymm commented on src/controllers/local-storage-page-controller.js:
```diff
@@ -107,8 +96,8 @@ angular.module('ToDo').controller('localStoragePageController',
-                            res[1] = JSON.parse(localStorage.getItem('allTodos'))[1];
+                            res[0] = todos;
+                            res[1] = completetodos;
                             $scope.mergeInService($scope.userChoise, res);
```
Looks like you don't need to use res.
```
 $scope.mergeInService($scope.userChoise, [todos. completetodos]);
```

BloamBla committed:
after reviw

- also some destructuring;
- some bug fixing.
